test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -175,21 +187,91 @@ class db_size_statistic : public statistic {
         file_names.push_back(std::string(DEFAULT_DIR) + "/" + WT_HS_FILE);
         file_names.push_back(std::string(DEFAULT_DIR) + "/" + WT_METAFILE);
 
-        return file_names;
+        return (file_names);
     }
 
     database &_database;
     int64_t _limit;
 };
 
+class postrun_statistic_check {
+    public:
+    explicit postrun_statistic_check(configuration *config)
+    {
+        const auto config_stats = config->get_list(POSTRUN_STATISTICS);
+        /*
+         * Each stat in the configuration is a colon separated list in the following format:
+         * <stat_name>:<min_limit>:<max_limit>
+         */
+        for (const auto &c : config_stats) {
+            auto stat = split_string(c, ':');
+            if (stat.size() != 3)
+                testutil_die(EINVAL,
+                  "runtime_monitor: Each postrun statistic must follow the format of "
+                  "\"stat_name:min_limit:max_limit\". Invalid format \"%s\" provided.",
+                  c.c_str());
+            const int min_limit = std::stoi(stat.at(1)), max_limit = std::stoi(stat.at(2));
+            if (min_limit > max_limit)
+                testutil_die(EINVAL,
+                  "runtime_monitor: The min limit of each postrun statistic must be less than or "
+                  "equal to its max limit. Config=\"%s\" Min=%ld Max=%ld",
+                  c.c_str(), min_limit, max_limit);
+            _stats.emplace_back(std::move(stat.at(0)), min_limit, max_limit);
+        }
+    }
+    virtual ~postrun_statistic_check() = default;
+
+    void
+    check(scoped_cursor &cursor) const
+    {
+        bool success = true;
+        for (const auto &stat : _stats)
+            success &= check_stat(cursor, stat);
```

Comment:
Bitwise operators on bool can result in unexpected behaviour in C, is that a concern in C++?

Reply:
If the answer is maybe or similar just use `success = success && check_stat(cursor, stat);`

Reply:
Additionally this will prevent unnecessary calls to check_stat while the previous implementation wont.

Reply:
We specifically want to check all of the stats even if one of them has failed as per the thread [here](https://github.com/wiredtiger/wiredtiger/pull/6739#discussion_r662051053).

It's apparently safe in C++ since `bool` is an actual type that can only be 0 or 1 (as opposed to C where it's just typedefs to an int). But I think this is a bit confusing and I worry that it will get flagged by Coverity or something since there's some ambiguity here. I'm thinking I'll use something like this:

```
for (const auto &stat : _stats) {
    if (!check_stat(cursor, stat)
        success = false;
}
if (!success)
     ...
```

Reply:
Yep sounds good, sorry missed the other thread.

Reply:
032ed45